Overriding CMAKE_CXX_FLAGS_DEBUG may cause link error in MSVC

When I was trying to create a `xeus` port (https://github.com/Microsoft/vcpkg/pull/5351) for `vcpkg`, I encountered a link error caused by the overriding of `CMAKE_CXX_FLAGS_DEBUG`.

### How to reproduce?

1. Install `vcpkg` on Win10, and install all packages needed by `xeus` via `vcpkg`:

   >vcpkg install cppzmq cryptopp

2. Clone `xeus` and try to build it with `vcpkg`'s cmake toolchain:

    ```bash
    cd xeus
    mkdir build
    cd build
    cmake .. -DCMAKE_TOOLCHAIN_FILE=E:/vcpkg/scripts/buildsystems/vcpkg.cmake -DVCPKG_TARGET_TRIPLET=x86-windows -DCMAKE_GENERATOR_PLATFORM=Win32 -DXEUS_USE_SHARED_CRYPTOPP=OFF -DCMAKE_BUILD_TYPE=Debug
   ```

3. Open the VS solution generated in previous step and build the `xeus` project (Win32 Debug):

   ```plain-text
    1>cryptopp-static.lib(cryptlib.cpp.obj) : error LNK2038: mismatch detected for '_ITERATOR_DEBUG_LEVEL': value '2' doesn't match value '0' in xauthentication.obj
    1>cryptopp-static.lib(cryptlib.cpp.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MDd_DynamicDebug' doesn't match value 'MD_DynamicRelease' in xauthentication.obj
    1>cryptopp-static.lib(misc.cpp.obj) : error LNK2038: mismatch detected for '_ITERATOR_DEBUG_LEVEL': value '2' doesn't match value '0' in xauthentication.obj
    1>cryptopp-static.lib(misc.cpp.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MDd_DynamicDebug' doesn't match value 'MD_DynamicRelease' in xauthentication.obj
   ......
   ```

If I change the `Runtime Library` from `Multi-threaded DLL (/MD)` to `Multi-threaded Debug DLL (/MDd)`(that is not overriding the default `CMAKE_CXX_FLAGS_DEBUG`) in the `xeus` project Properties dialog, this link error will disapper.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -230,12 +230,6 @@ if(MSVC)
     target_compile_definitions(xeus PUBLIC -DNOMINMAX)
     target_compile_options(xeus PUBLIC /DGUID_WINDOWS /MP /bigobj)
     target_compile_options(xeus PUBLIC /wd4251 /wd4996)
-    # Allows to compile in debug without requiring each dependencies to
-    # be compiled in debug
-    if(${U_CMAKE_BUILD_TYPE} MATCHES DEBUG)
-        set(CMAKE_CXX_FLAGS_DEBUG "/DNDEBUG /MD /Zi /Ob0 /Od /RTC1")
-        message(STATUS "Overriding CMAKE_CXX_FLAGS_DEBUG: ${CMAKE_CXX_FLAGS_DEBUG}")
-    endif()
 elseif(APPLE)
     target_compile_definitions(xeus PUBLIC -DGUID_CFUUID)
 else()

